### PR TITLE
optimize: remove H2 and pgsql get primary index code and close resultSet

### LINKED ISSUE #1534: Optimize get TableMeta
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
Optimize get TableMeta:
1. Remove H2 and pgsql indexing logic
2. Close the resultSet after use

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- OS :
- Others:

### PR DESCRIPTION
**Pull request by slievrly**
### Ⅰ. Describe what this PR did

optimize: remove H2 and pgsql get primary index code and close resultSet

### Ⅱ. Does this pull request fix one issue?
fix #1534

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheOracle.java [hunk 1]
```diff
@@ -113,99 +112,97 @@ private static TableMeta fetchSchemeInDefaultWay(DataSource dataSource, String t
 
     private static TableMeta resultSetMetaToSchema(ResultSetMetaData rsmd, DatabaseMetaData dbmd, String tableName)
```
Comment (zjinlei) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheOracle.java]: rsmd never used

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheOracle.java [hunk 2]
```diff
@@ -89,8 +89,7 @@ private static TableMeta fetchSchema(DataSource dataSource, String tableName) th
         return fetchSchemeInDefaultWay(dataSource, tableName);
     }
 
-    private static TableMeta fetchSchemeInDefaultWay(DataSource dataSource, String tableName)
-        throws SQLException {
+    private static TableMeta fetchSchemeInDefaultWay(DataSource dataSource, String tableName) throws SQLException {
         Connection conn = null;
```
Comment (zjinlei) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheOracle.java]: close conn

Comment (slievrly) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheOracle.java]: great found.

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheOracle.java (hunk 1 above)
Comment (slievrly) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheOracle.java]: fixed

**Pull request merged**